pkg/dash/workflow.go

```diff
@@ -89,15 +100,37 @@ func Apply(config Config, jsonnetFile string) error {
 	return nil
 }
 
-func renderDashboards(jsonnetFile string) (Boards, error) {
-	template := `
-  local f = import "{{FILE}}";
-  {
-    [k]: { dashboard: f.grafanaDashboards[k], folderId: 0, overwrite: true}
-    for k in std.objectFields(f.grafanaDashboards)
-  }
-  `
-	jsonnet := strings.ReplaceAll(template, "{{FILE}}", jsonnetFile)
+func dashboardKeys(jsonnetFile string) ([]string, error) {
+	jsonnet := fmt.Sprintf(`
+local f = import "%s";
+std.objectFields(f.grafanaDashboards)`, jsonnetFile)
+	output, err := evalToString(jsonnet)
+	if err != nil {
+		return nil, err
+	}
+	var keys []string
+	err = json.Unmarshal([]byte(output), &keys)
+	if err != nil {
+		return nil, err
+	}
+	return keys, nil
+}
```

Comment:
I wonder if the code might be smaller and easier to understand if `dashboardKeys` called renderDashboards, parsed the json and grabbed the keys from that. It would be slower, for sure, but exercises a single code path more and would I suspect be easier to read.

Reply:
I had considered that, but it assumes that all dashboards will compile. I went this route in case a user is ignoring certain dashboards because they are broken for whatever reason.